crates/net/network/src/transactions/mod.rs

```diff
@@ -1136,7 +1165,21 @@ where
 
             this.update_fetch_metrics();
 
-            // Advance all imports
+            // advance pool imports (flush txns to pool).
+            //
+            // note, this is done in batches. batches can potentially contain 20 k txns. a batch
+            // is filled from one `Transactions` broadcast messages or one `PooledTransactions`
+            // response at a time. each `PooledTransactions` responses could contain at least up
+            // to spec'd soft limit byte size for response / avg small legacy tx byte size: 2
+            // MiB / 100 bytes < 21 k transactions. the default soft limit byte size for
+            // `Transactions` broadcast messages is smaller, 128 KiB.
+            //
+            // the minimum batch size is 1 transaction. this is likely to occur since peers
+            // default to sending one blob per response (< 128 KiB).
+            //
+            // if txns however are invalid, and just 1 byte, since this isn't validated until
+            // import to pool, this can potentially queue > 2 million txns. more if the message
```

Comment:
nit: i'd add the "since [...]" part to the last part of the sentence i.e. "if txs [...], and just 1 byte, this can [..], since [..]", just reads more naturally